### DISCUSSION TITLE
Foxglove Studio takes too long to start

### DISCUSSION THREAD

**Discussion by swarajpeppermint** [foxglove/community#973] [Bugs]
**Description**
Since a few releases ago, Foxglove Studio takes too long to start up (white screen to home screen). Attaching a screen recording of the delay. During the screen recording, I just have the browser running on the system.

- Version: 1.82.0
- Platform: Ubuntu 22.04.3
- System Specs: RAM 16.0 GB, Processor 11th Gen Intel® Core™ i7-11800H @ 2.30GHz × 16
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native): N/A

**Steps To Reproduce**
Start Foxglove Studio on an updated Ubuntu 22.04 system

**Screen Recording**
[Screencast from 12-19-2023 09:43:18 AM.webm](https://github.com/foxglove/studio/assets/85288063/8c04d904-3bac-47a7-b386-035a4e3cf69a)

**Expected Behavior**
Should show home screen within a second or two

**Comment by foxhubber** [comment]
Internal tracking ticket: [FG-6142](https://linear.app/foxglove/issue/FG-6142/foxglove-studio-takes-too-long-to-start-[foxglovestudio-7260])

**Comment by jtbandes** [comment]
Can you press ctrl+shift+i when the window appears and see if there is any information in the developer console or network tab?

**Comment by swarajpeppermint** [comment]
![image](https://github.com/foxglove/studio/assets/85288063/268f92f7-f6ee-4575-afce-c83ef8874603)

**Comment by swarajpeppermint** [comment]
Foxglove Studio started hanging after a while
![image](https://github.com/foxglove/studio/assets/85288063/7ad50279-0391-41e2-a6a8-176759217b3b)

**Comment by swarajpeppermint** [comment]
network tab
![image](https://github.com/foxglove/studio/assets/85288063/0f61ce85-a636-4927-abac-407c10370318)

**Comment by wimagguc** [comment]
@swarajpeppermint sorry about the long wait, but we're looking into this at last. Unfortunately I'm having a hard time replicating the issue you described here — on my setup `1.72`, `1.82` and `1.87` all seem to cold-start within about the same time.

I also see in your console logs that a survey script wasn't loaded, and I wonder whether this has been a rather intermittent issue where a 3rd party integration slowed us down. Do you still see the issue on your end with the latest app releases?

**Comment by wimagguc** [comment]
Another question @swarajpeppermint: do you have any extensions installed? (Probably under `~/.foxglove-studio/extensions` or wherever snap has those.) Thanks for the question @achim-k

**Comment by playerdefault** [comment]
@wimagguc Yes there are 2 custom extensions installed. (@swarajpeppermint is my other username) They are: 
- https://github.com/playerdefault/foxglove-pointstamped-viz
- danielclapp.ros2-parameter-extension-0.0.1

**Comment by playerdefault** [comment]
A good update on this. Launching foxglove studio via `foxglove-studio --disable-gpu --disable-software-rasterizer` launches it **instantly**. The same issue is happening with VSCode too. 
Relevant GitHub issue with VSCode: https://github.com/microsoft/vscode/issues/192493#issuecomment-1937098984

Maybe it's not an issue with FoxGlove but something to do with a recent update with Electron?

**Comment by defunctzombie** [comment]
Folks experiencing this issue - did you install via Snap or Deb?

**Comment by playerdefault** [comment]
> Folks experiencing this issue - did you install via Snap or Deb?

snap

**Comment by defunctzombie** [comment]
@playerdefault if you are still experiencing this with the snap install - could you give the Deb install a try and see if it exhibits the same behavior on your machine. That would be a helpful data point for us.

**Comment by playerdefault** [comment]
> @playerdefault if you are still experiencing this with the snap install - could you give the Deb install a try and see if it exhibits the same behavior on your machine. That would be a helpful data point for us.

issue is not occurring with the deb version

**Comment by defunctzombie** [comment]
Anyone experiencing this I recommend using the _deb_ version. I suspect the snap version has some delays loading the app code related to how snap wants to launch files in the sandbox environment.
